src/http.rs

```diff
@@ -69,3 +90,107 @@ pub fn follow_redirects(
         Either::B(Ok(response).into_future())
     }))
 }
+
+/// A Resolver-like actor for the actix-web http client that refuses to resolve to reserved IP addresses.
+pub struct SafeResolver(Addr<Resolver>);
+
+impl Default for SafeResolver {
+    fn default() -> Self {
+        SafeResolver(Resolver::default().start())
+    }
+}
+
+impl Actor for SafeResolver {
+    type Context = Context<Self>;
+}
+
+impl Handler<Connect> for SafeResolver {
+    type Result = ResponseFuture<TcpStream, ResolverError>;
+
+    fn handle(&mut self, msg: Connect, _ctx: &mut Self::Context) -> Self::Result {
+        let Connect {
+            name,
+            port,
+            timeout,
+        } = msg;
+
+        let fut = self
+            .0
+            .send(Resolve { name, port })
+            .map_err(|mailbox_e| ResolverError::Resolver(mailbox_e.to_string()))
+            .flatten()
+            .and_then(move |mut addrs| {
+                addrs.retain(|addr| {
+                    for network in &*RESERVED_IP_BLOCKS {
+                        if network.contains(addr.ip()) {
+                            metric!(counter("http.blacklisted_ip") += 1);
+                            log::error!(
+                                "Blocked attempt to connect to reserved IP address: {}",
+                                addr.ip()
```

Comment:
Let's not log a error here, as this is very much expected to happen. The metric is A+ though.

Reply:
I think we want to make it visible when a user tries to screw with us like that, also there are usability issues that come from having security by default that I want to work around with by using an error log (user tries symbolicator out locally and wonders why their local bucket doesn't receive any requests)

Reply:
Discussed f2f, user needs to look into debug logs anyway if files could not be fetched (because it can be a lot of things that went wrong, not just this)